### DISCUSSION TITLE
Can't use REST endpoints

### DISCUSSION THREAD

**Discussion by zhamid-src** [supabase/supabase#19519] [Questions]
I am having trouble getting data back from REST endpoints. I followed these instructions: [Build an API route in less than 2 minutes. | Supabase Docs](https://supabase.com/docs/guides/api/quickstart?database-method=sql).

It gives me this error:

`{"code":"42501","details":null,"hint":null,"message":"permission denied for table todos"}`

This is how I am accessing it (copying from the documentation) :-
```
curl 'https://<PROJECT_REF>.supabase.co/rest/v1/todos' \
-H "apikey: <ANON_KEY>" \
-H "Authorization: Bearer <ANON_KEY>"
```
I used my project_ref and key.

To test, I changed the table name and got a proper relation not found error. I changed the API key and got the correct wrong API key error. 

I did the following to allow anon access. I also tried doing it using the dashboard and a postgres cli. No difference.

```
-- Turn on security
alter table "todos"
enable row level security;

-- Allow anonymous access
create policy "Allow anonymous access"
  on todos
  for select
  to anon
  using (true);
```
<img width="386" alt="image" src="https://github.com/supabase/supabase/assets/52141733/5599ad50-d7e3-4285-bd15-647f602fdaf6">

Using the service role key returns the same error.  

I can't figure out what's wrong. **Is it somehow SSL-related?**

**Comment by GaryAustin1** [comment]
Have you used something like Prisma on this database?  Seems something has messed with the grants on tables.
